tirocks-sys/src/lib.rs

```diff
@@ -13,20 +13,196 @@ mod bindings {
     include!(env!("BINDING_PATH"));
 }
 
+use std::{ffi::CStr, str::Utf8Error};
+
 pub use bindings::*;
 
-impl crocksdb_slice_t {
-    pub fn to_bytes(&self) -> &[u8] {
-        unsafe { std::slice::from_raw_parts(self.data as _, self.size) }
+/// Convert a rust slice to rocksdb slice.
+///
+/// # Safety
+///
+/// Caller should guarantee the data lives as long as the returned slice.
+// Because the layout of rocksdb_Slice is exact the same as &[u8], this function
+// will be optimized away by compiler.
+#[inline]
+pub unsafe fn r(src: &[u8]) -> rocksdb_Slice {
+    rocksdb_Slice {
+        data_: src.as_ptr() as _,
+        size_: src.len(),
     }
 }
 
+/// Convert a rocksdb slice to rust slice.
+///
+/// # Safety
+///
+/// Caller should guarantee the data lives as long as the returned slice.
+// Because the layout of rocksdb_Slice is exact the same as &[u8], this function
+// will be optimized away by compiler.
+#[inline]
+pub unsafe fn s<'a>(src: rocksdb_Slice) -> &'a [u8] {
+    std::slice::from_raw_parts(src.data_ as _, src.size_)
+}
+
+impl rocksdb_Status {
+    #[inline]
+    pub fn with_code(code: rocksdb_Status_Code) -> rocksdb_Status {
+        rocksdb_Status {
+            code_: code,
+            subcode_: rocksdb_Status_SubCode::kNone,
+            sev_: rocksdb_Status_Severity::kNoError,
+            state_: std::ptr::null(),
+        }
+    }
+
+    /// Build a status with given errors.
+    ///
+    /// # Panics
+    ///
+    /// `code` should not be `kOK`.
+    #[inline]
+    pub fn with_error(code: rocksdb_Status_Code, state: impl AsRef<[u8]>) -> rocksdb_Status {
```

Comment:
in which case it would be used?

Reply:
For example, return error status from rust code to C++ code. https://github.com/BusyJay/tirocks/blob/78357252ba6eb09e42865b855aca3de4c2fb6bb0/tirocks-sys/crocksdb/c.cc#L4341